experiments/coredns/pod_delete/pod_delete_ansible_logic.yml

```diff
@@ -0,0 +1,119 @@
+        - block:
+            - name: Construct chaos result name (experiment_name)
+              set_fact:
+                c_experiment: "{{ lookup('env','CHAOSENGINE') }}-{{ c_experiment }}"
+
+          when: lookup('env','CHAOSENGINE')    
+
+        ## RECORD START-OF-EXPERIMENT IN LITMUSCHAOS RESULT CR
+        - include_tasks: /utils/runtime/update_chaos_result_resource.yml
+          vars:
+            status: 'SOT'
+            namespace: "{{ c_ns }}"
+        
+        ## Provision services and pods for the chaos injection
+        - name: Provision nginx pod and liveness pod
+          block:
+            - template:
+                src: nginx.j2
+                dest: nginx.yml
+            
+            - template:
+                src: liveness-app.j2
+                dest: liveness-app.yml
+              
+            - name: Deploying and exposing nginx pod
+              shell: kubectl apply -f nginx.yml
+            
+            - name: Deploying liveness pod 
+              shell: kubectl apply -f liveness-app.yml
+
+        ## PRE-CHAOS APPLICATION LIVENESS CHECK
+        - name: Pre-chaos application liveness check
+          block:
+            - name: Verify that the AUT (Application Under Test- coredns app) is running 
+              include_tasks: "/utils/common/status_app_pod.yml"
+              vars:     
+                delay: 1
+                retries: 60
+
+            - name: Verify that the AUT (Application Under Test- liveness app) is running 
+              include_tasks: "/utils/common/status_app_pod.yml"
+              vars:     
+                delay: 1
+                retries: 60
+                a_ns: "{{ c_ns }}"
+                a_label: "run=liveness-app"
+  
+        ## FAULT INJECTION 
+        - include_tasks: "{{ c_util }}"
+          vars:
+            app_ns: "{{ a_ns }}"
+            app_label: "{{ a_label }}"
+          
+        ## POST-CHAOS APPLICATION LIVENESS CHECK
+        - name: Pre-chaos application liveness check
+          block:
+            - name: Verify that the AUT (Application Under Test- coredns app) is running 
+              include_tasks: "/utils/common/status_app_pod.yml"
+              vars:     
+                delay: 1
+                retries: 60
+
+            - name: Verify that the AUT (Application Under Test- liveness app) is running 
+              include_tasks: "/utils/common/status_app_pod.yml"
+              vars:     
+                delay: 1
+                retries: 60
+                a_ns: "{{ c_ns }}"
+                a_label: "run=liveness-app"
+
+        ## DEPROVISIONING TASK
+        - name: Cleaning up chaos pods and services
+          block: 
+            - name: deleting service and pod of nginx
+              shell: kubectl delete -f nginx.yml
+            
+            - name: deleting liveness pod 
+              shell: kubectl delete -f liveness-app.yml
+
+        - set_fact:
+            flag: "pass"
+
+      rescue:
```

Comment:
lets have a failsafe by checking whether nginx/liveness is deployed and cleaning up if present

Reply:
done